src/jit/lowerxarch.cpp

```diff
@@ -2409,24 +2409,29 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
                 break;
         }
     }
-    else if (numArgs == 3)
+
+    if (Compiler::categoryOfHWIntrinsic(intrinsicID) == HW_Category_IMM)
```

Comment:
Why `if` instead of `else if`?

Reply:
Nevermind, to handle the two-op case (such as shifting).

we might want to return, instead of break, on a couple of the cases above, since they are known to not have the second-operand be imm.

Reply:
Because imm-intrinsics can have 2 operands.